pl_examples/hydra_examples/pl_hydra/trainer_conf.py

```diff
@@ -0,0 +1,257 @@
+from dataclasses import dataclass, asdict
+from typing import Optional, List, Dict, Any
+
+from hydra.utils import instantiate
+from hydra.core.config_store import ConfigStore
+from hydra.types import ObjectConf
+
+
+cs = ConfigStore.instance()
+
+
+@dataclass
+class LightningTrainerConf:
+    # callbacks: Optional[List[Callback]] = None
```

Comment:
I think we are ready to start tackling left-over todos.

Reply:
So this one, I feel we would need some user help. For example, how do we support the list of callback objects? 1) A user would need to define a structured config for their callback, 2) They would need to create a yaml list of these configs, 3) We would need to instantiate their object and populate the list, 4) Pass this into trainer.

Reply:
Taking the callback from the example:
```python
class MyPrintingCallback(Callback):

    def on_init_start(self, trainer):
        print('Starting to init trainer!')

    def on_init_end(self, trainer):
        print('trainer is init now')

    def on_train_end(self, trainer, pl_module):
        print('do something when training ends')
```

This style lets you reuse the callback configs more easily:
```yaml
callbacks:
  print:
    cls: MyPrintingCallback
  s3_checkpoint:
    cls: S3Checkpoint
    params:
      bucket_name: ???

callbacks_list:
  - ${callbacks.print}
  - ${callbacks.checkpoint}
```

The code can instantiate the callbacks like:
```python
callbacks = [hydra.utils.instantiate(callback) for callback in cfg.callbacks_list]
```

Currently, Config group are mutually exclusive: you can only load one config from each config group.
Once https://github.com/facebookresearch/hydra/issues/499 is done we will be able to do something better.
For now you the example can put all the callbacks config in the primary config file or break them into *callbacks.yaml* that can be added to the defaults list as `- callbacks`.

PS: I did not try this so there may be unforeseen problems.

Reply:
Question about interpolation, when I define callbacks: ${callbbacks.print}, then this resolves to: {'cls': 'MyPrintingCallback'}. Although when I put callbacks into a list format it resolves to {'callbacks': ['${callbacks.print}']}. Is there anything special you need to do for variable interpolation in a list?

Reply:
Okay, I figured out the above issue. The solution you mentioned did not work, I was getting an error regarding incorrect type when trying to use the object:  cfg.callbacks_list[0]. I put up a working solution in the most recent commit by just listing out the names of these fields.

Reply:
What error? can you be more specific?

Reply:
Nevermind, I fixed the example and now it works like your example.